configure.ac

```diff
@@ -381,6 +381,10 @@ AC_ARG_WITH(priv-mode,
 
 AM_CONDITIONAL(PRIV_MODE_SETUID, test "x$with_priv_mode" = "xsetuid")
 
+if test "x$with_priv_mode" = "xsetuid"; then
```

Comment:
I'm not a fan of having the "it's setuid" and "it's unprivileged" builds be functionally different - that would invalidate the instructions we give to Debian users (where bubblewrap was historically setuid).

I'd prefer to stat the bubblewrap executable and check for `st.st_mode & S_ISUID`, like we already do in `portal/flatpak-portal.c`. That way, the same build can be used either way.

Reply:
I was thinking that we'd want to avoid the risk of the "normal" (non-setuid) case ever erroneously thinking it needed to disable this feature by making that a compile time thing. However, I can see your reasoning too, and I'm fine with that.

Reply:
> I was thinking that we'd want to avoid the risk of the "normal" (non-setuid) case ever erroneously thinking it needed to disable this feature

If we want that, I'd prefer it to be a separate feature, `--with-unprivileged-userns-required` or something like that, with its effect looking something like this during startup:

```
#ifdef UNPRIVILEGED_USERNS_REQUIRED
  if (bwrap_is_setuid)
    error ("This build of Flatpak requires a non-setuid bwrap executable, but %s is setuid", bwrap);
#endif
```

so that vendors who know that their OS's only supported configuration is the unprivileged one can guarantee that features like `--disable-userns` and `FLATPAK_SPAWN_FLAGS_EXPOSE_PIDS` will work.

A similar option in bubblewrap, to make it error out if it's setuid, might also make sense (and if Flatpak is asked to require the non-setuid bubblewrap and is compiling its bundled copy, it should pass on the option).

However, I think the default should be that the same Flatpak and bubblewrap builds can be used either way (until we stop caring about Debian 10, RHEL 7 and "hardened" Arch kernels, at which point we can flip the default or even make it always-on).

For the builds in Debian, I didn't use `--with-priv-mode=setuid` even when we were making bubblewrap setuid, because none of the process of building a Debian package happens as root anyway - the earliest point at which a Debian package has the ability to do arbitrary things as root is the `preinst`/`postinst` scripts that run when you install it.

Reply:
I'm fine with just checking the setuid bit. Its not a very serious risk anyway, and you'd have to be root to make a setuid bit to fool the check.